app/src/main/java/edu/agh/susgame/front/providers/interfaces/AwaitingGamesProvider.kt

```diff
@@ -12,4 +12,6 @@ interface AwaitingGamesProvider {
     fun join(id: GameId, playerNickname: PlayerNickname)
 
     fun leave(id: GameId, playerNickname: PlayerNickname)
+
+    fun createNewGame(gameId: GameId, gameName: String, gamePIN: String, numOfPlayers: Int, gameTime: Int)
```

Comment:
Myślę, że więcej sensu by miało, żeby ta metoda nie przyjmowała `GameId` jako argumentu, a zwracała `GameId` nowo utworzonej gry. Wtedy w tej mockowej implementacji `MockAwaitingGamesProvider` mógłbyś sobie stworzyć takie `var freeId: Int`, ale już prywatne

Reply:
Trochę się nie zrozumieliśmy, chodziło mi o coś takiego:
(aby to wrzucić, skopiuj całość i  `crtl + shift + a` => `Apply Patch from Clipboard`)
```
Index: app/src/main/java/edu/agh/susgame/front/model/game/AwaitingGame.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/edu/agh/susgame/front/model/game/AwaitingGame.kt b/app/src/main/java/edu/agh/susgame/front/model/game/AwaitingGame.kt
--- a/app/src/main/java/edu/agh/susgame/front/model/game/AwaitingGame.kt	(revision d33f4fa41885462cf65530a73c3af773f9447315)
+++ b/app/src/main/java/edu/agh/susgame/front/model/game/AwaitingGame.kt	(date 1715552889939)
@@ -9,10 +9,4 @@
     val gameTime: Int,
     val gamePin: String, // probably not safe here, should be hidden in the future
     val playersWaiting: List<PlayerNickname>,
-
-    ) {
-    companion object {
-        var freeId: Int =
-            0 // current free id for new game, variable is static so it's shared among all instances
-    }
-}
+)
Index: app/src/main/java/edu/agh/susgame/front/providers/mock/MockAwaitingGamesProvider.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/edu/agh/susgame/front/providers/mock/MockAwaitingGamesProvider.kt b/app/src/main/java/edu/agh/susgame/front/providers/mock/MockAwaitingGamesProvider.kt
--- a/app/src/main/java/edu/agh/susgame/front/providers/mock/MockAwaitingGamesProvider.kt	(revision d33f4fa41885462cf65530a73c3af773f9447315)
+++ b/app/src/main/java/edu/agh/susgame/front/providers/mock/MockAwaitingGamesProvider.kt	(date 1715552889959)
@@ -8,9 +8,12 @@
 import edu.agh.susgame.front.settings.Configuration
 
 class MockAwaitingGamesProvider : AwaitingGamesProvider {
+    // Current free id for new game
+    private var freeGameId: Int = 0
+
     private var state = mutableListOf(
         AwaitingGame(
-            id = GameId(AwaitingGame.freeId++),
+            id = GameId(freeGameId++),
             name = "Gra dodana statycznie",
             maxNumOfPlayers = Configuration.MaxPlayersPerGame,
             gameTime = 10,
@@ -49,7 +52,7 @@
         numOfPlayers: Int,
         gameTime: Int
     ): GameId {
-        val gameId = GameId(AwaitingGame.freeId++)
+        val gameId = GameId(freeGameId++)
         state.add(
             AwaitingGame(
                 gameId,
@@ -57,10 +60,9 @@
                 numOfPlayers,
                 gameTime,
                 gamePIN,
-                emptyList<String>().map { PlayerNickname(it) }
+                emptyList(),
             )
         )
         return gameId
     }
-
-}
\ No newline at end of file
+}
```